gtsam/hybrid/HybridBayesNet.cpp

```diff
@@ -43,4 +47,78 @@ GaussianBayesNet HybridBayesNet::choose(
   return gbn;
 }
 
+/* ************************************************************************* */
+HybridBayesNet::shared_ptr HybridBayesNet::prune(
+    const DecisionTreeFactor::shared_ptr &discreteFactor) const {
+  // To Prune, we visitWith every leaf in the GaussianMixture. For each
+  // leaf, we apply an operation, where using the assignment, we can
+  // check the discrete decision tree for an exception and if yes, then just
+  // set the leaf to a nullptr. We can later check the GaussianMixture for
+  // just nullptrs.
+
+  HybridBayesNet::shared_ptr prunedBayesNetFragment =
+      boost::make_shared<HybridBayesNet>(*this);
+
+  // Go through all the conditionals in the
+  // Bayes Net and prune them as per discreteFactor.
+  for (size_t i = 0; i < this->size(); i++) {
+    auto conditional = this->at(i);
+
+    // Container for nodes (or nullptrs) to create a new DecisionTree
+    std::vector<GaussianConditional::shared_ptr> nodes;
+
+    // Loop over all assignments and create a set of GaussianConditionals
+    std::function<GaussianFactor::shared_ptr(
```

Comment:
auto?

Reply:
I kind of like using the full type signature since it helps inform the lambda definition (aka self-documenting).

Reply:
but it is right there in the argument list?... Add return type via `->` if you are talking about the return type not being clear.

Reply:
I was thinking about the tooltip in VS Code but I'll update it.